src/unk_02073C2C.c

```diff
@@ -111,211 +113,207 @@ static const s8 Unk_020F0695[][5] = {
+void sub_02077F0C(Pokemon * param0, u32 param1, int param2);
+BOOL sub_02077FB4(Pokemon * param0, u8 param1);
+BOOL sub_02077FBC(BoxPokemon * param0, u8 param1);
 BOOL sub_02077FE4(u16 param0, int param1, u8 param2);
-void sub_0207803C(UnkStruct_02073C74 * param0);
-void sub_02078044(UnkStruct_02073C74_sub1 * param0);
-void sub_020780C4(UnkStruct_02073C74 * param0, u32 param1);
+void sub_0207803C(Pokemon * param0);
+void sub_02078044(BoxPokemon * param0);
+void sub_020780C4(Pokemon * param0, u32 param1);
 static void sub_020781F8(int param0, UnkStruct_02075874 * param1);
 static void sub_02078208(int param0, int param1, UnkStruct_02075874 * param2);
 static void sub_0207821C(int param0, UnkStruct_0207821C * param1);
 static void sub_0207822C(void * param0, u32 param1, u32 param2);
 static void sub_02078234(void * param0, u32 param1, u32 param2);
 static u16 sub_0207823C(void * param0, u32 param1);
-static void * sub_0207825C(UnkStruct_02073C74_sub1 * param0, u32 param1, u8 param2);
+static void * sub_0207825C(BoxPokemon * param0, u32 param1, u8 param2);
 static int sub_02078740(int param0, int param1);
 u32 sub_020787CC(int param0);
 int sub_020787EC(u32 param0);
 BOOL sub_02078804(u16 param0);
-BOOL sub_02078838(UnkStruct_02073C74 * param0);
+BOOL sub_02078838(Pokemon * param0);
 int sub_020788D0(int param0);
 static void sub_02076300(UnkStruct_02008A90 * param0, u16 param1, u8 param2, u8 param3, u8 param4, u8 param5, u32 param6);
 static u8 sub_020767BC(u16 param0, u8 param1, u8 param2, u8 param3, u32 param4);
 
-void sub_02073C2C (UnkStruct_02073C74 * param0)
+void ZeroMonData (Pokemon * mon)
 {
-    MI_CpuClearFast(param0, sizeof(UnkStruct_02073C74));
-
-    sub_0207822C(&param0->unk_00.unk_08, sizeof(UnkStruct_02075454) * 4, param0->unk_00.unk_06);
-    sub_0207822C(&param0->unk_08, sizeof(UnkStruct_02073C74_sub2), param0->unk_00.unk_00);
+    MI_CpuClearFast(mon, sizeof(Pokemon));
+    sub_0207822C(&mon->box.unk_08, sizeof(PokemonSubstruct0) * 4, mon->box.unk_06);
+    sub_0207822C(&mon->party, sizeof(PartyPokemon), mon->box.unk_00);
 }
 
-void sub_02073C54 (UnkStruct_02073C74_sub1 * param0)
+void ZeroBoxMonData (BoxPokemon * boxMon)
 {
-    MI_CpuClearFast(param0, sizeof(UnkStruct_02073C74_sub1));
-    sub_0207822C(param0->unk_08, sizeof(UnkStruct_02075454) * 4, param0->unk_06);
+    MI_CpuClearFast(boxMon, sizeof(BoxPokemon));
+    sub_0207822C(boxMon->unk_08, sizeof(PokemonSubstruct0) * 4, boxMon->unk_06);
 }
 
-int sub_02073C70 (void)
+int SizeOfStructPokemon (void)
 {
-    return sizeof(UnkStruct_02073C74);
+    return sizeof(Pokemon);
 }
 
-UnkStruct_02073C74 * sub_02073C74 (u32 param0)
+Pokemon * AllocMonZeroed (u32 param0)
 {
-    UnkStruct_02073C74 * v0;
-
-    v0 = Heap_AllocFromHeap(param0, sizeof(UnkStruct_02073C74));
-    sub_02073C2C(v0);
-
-    return v0;
+    Pokemon * mon = Heap_AllocFromHeap(param0, sizeof(Pokemon));
```

Comment:
Not sure but maybe CallocMon? This diverges from gen 3 naming tho

Reply:
Hmm, I feel we can find ways to improve over Gen 3's documentation by applying more proper names now and then backporting those changes to the Gen 3 repos.